src/lib/Microsoft.Health.Events/EventHubProcessor/EventProcessor.cs

```diff
@@ -111,5 +113,18 @@ async Task ProcessInitializingHandler(PartitionInitializingEventArgs initArgs)
                 processor.PartitionInitializingAsync -= ProcessInitializingHandler;
             }
         }
+
+        private (Exception exception, string reason) ProcessErrorEventException(Exception exception)
```

Comment:
Would it be better to have our own exception type for these scenarios?  We can use the original exception as the inner exception but have a strong typed exception class with error codes and other properties.

Reply:
I feel like bulk of the event hub processing exceptions are captured in EventHubsException with explicit reason/error  [codes](https://docs.microsoft.com/en-us/dotnet/api/azure.messaging.eventhubs.eventhubsexception.failurereason?view=azure-dotnet) and other properties like IsTransient, EventHubName, Message etc. Its just the invalid host which isn't captured in here.
The only thing I can think of to wrap in our own exception class would be if we further want to classify the exceptions from our perspective of IsClient/Service - eg: if invalid event hub was provided then its a client error for us, etc.
If we see value in that, then can add a new exception type.